mongodbatlas/continuous_snaphots.go

```diff
@@ -46,16 +46,19 @@ type ContinuousSnapshot struct {
 }
 
 type Part struct {
-	ClusterID          string `json:"clusterId"`
-	CompressionSetting string `json:"compressionSetting"`
-	DataSizeBytes      int64  `json:"dataSizeBytes"`
-	EncryptionEnabled  bool   `json:"encryptionEnabled"`
-	FileSizeBytes      int64  `json:"fileSizeBytes"`
-	MasterKeyUUID      string `json:"masterKeyUUID,omitempty"`
-	MongodVersion      string `json:"mongodVersion"`
-	ReplicaSetName     string `json:"replicaSetName"`
-	StorageSizeBytes   int64  `json:"storageSizeBytes"`
-	TypeName           string `json:"typeName"`
+	ClusterID          string            `json:"clusterId,omitempty"`
+	CompressionSetting string            `json:"compressionSetting,omitempty"`
+	DataSizeBytes      int64             `json:"dataSizeBytes,omitempty"`
+	EncryptionEnabled  bool              `json:"encryptionEnabled,omitempty"`
+	FileSizeBytes      int64             `json:"fileSizeBytes,omitempty"`
+	MasterKeyUUID      string            `json:"masterKeyUUID,omitempty"`
+	MongodVersion      string            `json:"mongodVersion,omitempty"`
+	ReplicaSetName     string            `json:"replicaSetName,omitempty"`
+	StorageSizeBytes   int64             `json:"storageSizeBytes,omitempty"`
+	TypeName           string            `json:"typeName"`
+	ShardName          string            `json:"shardName,omitempty"`
+	TokenDiscovered    bool              `json:"tokenDiscovered,omitempty"`
+	TokenTimestamp     SnapshotTimestamp `json:"tokenTimestamp,omitempty"`
```

Comment:
Is you want to make this ommitempty you may need to make it a pointer

Reply:
Do you prefer making them pointers or creating a new struct? I am thinking about the latter

Reply:
if this is the same for both but with extra fields let's make them pointers

Reply:
another 3rd way is creating 3 structs, one base, one with only the extra and then one that compose both, composition sometimes is great ;)